### PR TITLE
Add user_factory

### PR DESCRIPTION
**Pull request by fbalak**
Users created by the factory will be used for Prometheus API communication - this will help to run monitoring tests on IBM cloud and other platforms that doesn't generate kubeadmin-password file.

### CODE REVIEW

**Review by ebenahar — DISMISSED** [review]

File: ocs_ci/utility/users.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        replace (bool): If secret already exists then this will replace it
+
+    """
+    kubeconfig = os.getenv('KUBECONFIG')
+    if replace:
+        replace = (
+            f" --dry-run -o yaml | oc replace --kubeconfig {kubeconfig} -f -"
+        )
+    else:
+        replace = ''
```
Comment (petr-balogh) [inline ocs_ci/utility/users.py:43]: I would rather use different variable than the argument one, something like
```suggestion
    replace_str = ''
    if replace:
        replace_str = (
            f" --dry-run -o yaml | oc replace --kubeconfig {kubeconfig} -f -"
        )
```

And of course than later on use replace_str

File: tests/libtest/test_user.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import logging
+
+from ocs_ci.utility.utils import exec_cmd
+
+logger = logging.getLogger(__name__)
+
+
+def test_bucket_creation(user_factory):
+    user = user_factory()
+    exec_cmd(['oc', 'login', '-u', user[0], '-p', user[1]])
```
Comment (petr-balogh) [inline tests/libtest/test_user.py:10]: do we want to mask the password in the logs?

File: ocs_ci/utility/users.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        username (str): Name of a new user
+        password (str): Password for a new user
+        htpasswd_path (str): Path to httpasswd file
+
+    """
+    if os.path.isfile(htpasswd_path):
+        cmd = ['htpasswd', '-B', '-b', htpasswd_path, username, password]
+    else:
+        cmd = ['htpasswd', '-c', '-B', '-b', htpasswd_path, username, password]
+    exec_cmd(cmd)
```
Comment (petr-balogh) [inline ocs_ci/utility/users.py:25]: Do we want to mask the password in the log? We have secrets parameter for this purpose:
https://github.com/red-hat-storage/ocs-ci/blob/c1abe2b69af1b4aabc7e04aa7e5f8c4939a094f5/ocs_ci/utility/utils.py#L419

And this is proper place where to use it I guess.

**Review by petr-balogh — COMMENTED** [review]
Otherwise LGTM

**Review by petr-balogh — APPROVED** [review]

File: tests/libtest/test_user.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import os
+from time import sleep
+
+from ocs_ci.utility.utils import exec_cmd
+
+logger = logging.getLogger(__name__)
+
+
+def test_user_creation(user_factory):
+    user = user_factory()
```
Comment (mbukatov) [inline tests/libtest/test_user.py:11]: Use `username, password = user_factory()` instead.

**Review by mbukatov — APPROVED** [review]

File: tests/libtest/test_user.py (hunk 4 above)
Comment (fbalak) [reply tests/libtest/test_user.py:11]: I will add it in next PR

**Pull request merged**